src/main/scala/model/Tree.scala

```diff
@@ -0,0 +1,43 @@
+package model
+
+/**
+ *  Represents a Tree.
+ */
+sealed trait BinaryTree[A]{
+  val elem: A
```

Comment:
```suggestion
  val leafElement: A
```

Reply:
Questo a me trarrebbe in inganno perchè abbiamo l'element anche nel Node, non solo nella Leaf

Reply:
Giusto ho commentato prima di revisionare Node. Dici che ci potrebbe essere un modo per distinguerli o lasciamo element e sarà il riferimento alla classe che ci indica se è di Node o di BynaryTree?

Reply:
Secondo deve essere lo stesso in modo che tu possa accedere a prescindere che sia un Leaf o un Node

Reply:
ok mi è chiaro. Va bene!